ERCS/erc-7649.md

```diff
@@ -0,0 +1,446 @@
+	 * @dev Depending on the implementation, ERC-20 token payment may be required instead of ETH.
+	 *      In such a case, implementation must through if ETH is sent, effectively overriding
+	 *      the function definition as non-payable
+	 *
+	 * @param amount amount of the shares to buy
+	 */
+	function buyShares(uint256 amount) external payable;
+
+	/**
+	 * @notice Buy `amount` of shares in the favor of the address specified (beneficiary).
+	 *      Sender has to supply `getBuyPriceAfterFee(amount)` ETH.
+	 *      First share can be bought only by current subject issuer.
+	 *
+	 * @dev Depending on the implementation, ERC-20 token payment may be required instead of ETH.
+	 *      In such a case, implementation must through if ETH is sent, effectively overriding
+	 *      the function definition as non-payable
+	 *
+	 * @param amount amount of the shares to buy
+	 * @param beneficiary an address receiving the shares
+	 */
+	function buySharesTo(uint256 amount, address beneficiary) external payable;
+
+	/**
+	 * @notice Sell `amount` of shares. Sender gets `getSellPriceAfterFee(amount)` of ETH.
+	 *      Last share cannot be sold.
+	 *
+	 * @dev Depending on the implementation, ERC-20 token may be payed instead of ETH.
+	 *
+	 * @param amount amount of the shares to sell
+	 */
+	function sellShares(uint256 amount) external;
+
+	/**
+	 * @notice Sell `amount` of shares in the favor of the address specified (beneficiary).
+	 *      The beneficiary gets `getSellPriceAfterFee(amount)` of ETH.
+	 *      Last share cannot be sold.
+	 *
+	 * @dev Depending on the implementation, ERC-20 token may be payed instead of ETH.
+	 *
+	 * @param amount amount of the shares to sell
+	 * @param beneficiary an address receiving the funds from the sale
+	 */
+	function sellSharesTo(uint256 amount, address payable beneficiary) external;
+
+	/**
+	 * @notice Cumulative value of all trades; allows to derive cumulative fees paid
+	 *
+	 * @dev This value cannot decrease over time; it can increase or remain constant
+	 *      if no trades are happening
+	 *
+	 * @return Sum of the modulo of all trading operations
+	 */
+	function getTradeVolume() external view returns(uint256);
+```
+
+
+## Rationale
+
+The rationale behind the design choices for the embedded liquidity standard is deeply rooted in providing a robust and
+versatile framework for embedding Bonding Curve-like liquidity into NFTs. The following key considerations have
+influenced the technical decisions:
+
+1.  **Bonding Curve-Embedded Liquidity / Tradable Shares Contract**:
+    - **Seamless Integration**: The decision to allow an embedded bonding curve-based liquidity contract to be attached
+        to an NFT without altering the ERC-721 standard stems from the desire for seamless integration.
+        This approach ensures that NFT developers can enhance their creations with liquidity mechanisms without
+        introducing complexities or requiring modifications to the widely adopted ERC-721 standard.
```

Comment:
This point repeats content from line ~69. Please move content explaining "why you did things" into the Rationale section, and only keep specific requirements in the Specification section.